types/koa-joi-router/index.d.ts

```diff
@@ -31,20 +46,22 @@ declare namespace createRouter {
         };
     }
 
-    interface Request extends Koa.Request {
-        body: any;
-        params: {[key: string]: string};
-    }
-
-    interface Context extends Koa.Context {
-        request: Request;
-    }
-
     interface Router {
         routes: Spec[];
         route(spec: Spec|Spec[]): Router;
         middleware(): Koa.Middleware;
-        prefix(path: string): void;
+        prefix(path: string): Router;
```

Comment:
`prefix` doesn't return `this`, it's delegated to the koa-router, which returns that router.

Same for `use` and `param`

https://github.com/koajs/joi-router/blob/master/joi-router.js#L42-L45

Reply:
Thank you for your review. I decided to reuse the koa-router type.